File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/controller/brandController.js
// controller 只和html交互 不直接和后台交互
app.controller("brandController",function($scope,$controller, brandService){  //直接把brandService注入到controller中

    // 控制器的继承
      $controller('baseController',{$scope:$scope});//继承   本质上是共用同一个$scope

        $scope.findPage=function(pageNum,pageSize){
            brandService.findPage(pageNum,pageSize).success(function (response) {
                // 返回的数据 ：总条数 和 当前页的数据 {total:110,rows:[{},{},{}]}
                $scope.paginationConf.totalItems = response.total;
                $scope.list = response.rows;
            })
        }

        $scope.findAll=function () {
            brandService.findAll().success(function(response){
                // response就是查询出的所有数据
                $scope.list = response;
            })
        }

        $scope.save=function () {
            var  obj = null;
            if($scope.entity.id!=null){
                obj = brandService.update($scope.entity);
            }else{
                obj = brandService.add($scope.entity);
            }
            obj.success(function (response) {
                if(response.success){
                    $scope.reloadList();
                }else{
                    alert(response.message);
                }
            });
        }

        $scope.findOne=function(id){
            brandService.findOne(id).success(function (response) {
                // response就应该是返回的一个品牌对象
                $scope.entity=response;
            })
        }


        $scope.dele=function(){
            // 判断数组是否为空 如果为空 return
            if($scope.selectedIds.length==0){
                return;
            }

            var flag = window.confirm("确认要删除您选择的数据吗?");
            if(flag){
                brandService.dele($scope.selectedIds).success(function (response) {
                    if(response.success){
                        $scope.reloadList();
                        // 把数组清空
                        $scope.selectedIds=[];
                    }else{
                        alert(response.message);
                    }
                })
            }
        }
        $scope.searchEntity={};//需要初始化 不然是undefined
        $scope.search=function(pageNum,pageSize){
            // 向后台传的参数有：$scope.searchEntity
            // 当前页码$scope.paginationConf.currentPage,每页显示的条数$scope.paginationConf.itemsPerPage
            brandService.search(pageNum,pageSize,$scope.searchEntity).success(function (response) {
                // 返回的数据 ：总条数 和 当前页的数据 {total:110,rows:[{},{},{}]}
                $scope.paginationConf.totalItems = response.total;
                $scope.list = response.rows;
            })

        }
    })

File: pinyougou_parent/pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

//@Controller
//@ResponseBody //转json 并且回显到浏览器
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;
    // 此方法是select2要求的  需要的格式 [{'id':1,'text':'aaaa'},{'id':2,'text':'bbbb'},{'id':3,'text':'cccc'}]
    @RequestMapping("/findBrandList")
    public List<Map> findBrandList(){
       return brandService.findBrandList();
    }
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
       return brandService.findAll();
    }
    @RequestMapping("/findPage")
    public PageResult findPage(int pageNum, int pageSize){
//        {total:110,rows:[{},{},{}]}
       return brandService.findPage(pageNum,pageSize);
    }
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){
//        {success:true|false,message:"保存成功"|"保存失败"}
       try {
           brandService.add(brand);
           return new Result(true,"保存成功");
       }catch (Exception e){
           e.printStackTrace();
           return new Result(false,"保存失败");
       }
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true,"修改成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }
    @RequestMapping("/dele")
    public Result dele(Long[] ids){
        try {
            brandService.dele(ids);
            return new Result(true,"删除成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return  brandService.findOne(id);
    }

    @RequestMapping("/search")
    public PageResult findPage(int pageNum, int pageSize,@RequestBody TbBrand brand){
//        {total:110,rows:[{},{},{}]}
        return brandService.search(pageNum,pageSize,brand);
    }

}
